StreamChatCoreTests/Unit Tests/ClientLoggerTests.swift

```diff
@@ -0,0 +1,168 @@
+                        .unflagMessage(testMessage),
+                        .users(.init(filter: testFilter)),
+                        .updateUsers([testUser]),
+                        .muteUser(testUser),
+                        .unmuteUser(testUser),
+                        .flagUser(testUser),
+                        .unflagUser(testUser),
+                        .ban(.init(user: testUser, channel: testChannel, timeoutInMinutes: 0, reason: nil))]
+    }
+    
+    func testLogEndpointStrings() {
+        for endpoint in allEndpoints {
+            logger.log("Endpoint: \(String(describing: endpoint))", level: .debug)
+        }
+        
+        XCTAssertEqual(logged.count, allEndpoints.count)
+    }
+    
+    func testLogURLRequest() {
+        for endpoint in allEndpoints {
+            var urlRequest = URLRequest(url: testUrl)
+            urlRequest.httpMethod = endpoint.method.rawValue
+            urlRequest.addValue("application/json", forHTTPHeaderField: "Content-Type")
+            
+            if let body = endpoint.body {
+                let encodable = AnyEncodable(body)
+                
+                do {
+                    if let httpBody = try? JSONEncoder.defaultGzip.encode(encodable) {
+                        urlRequest.httpBody = httpBody
+                        urlRequest.addValue("gzip", forHTTPHeaderField: "Content-Encoding")
+                    } else {
+                        urlRequest.httpBody = try JSONEncoder.default.encode(encodable)
+                    }
+                } catch {
+                    XCTFail(error.localizedDescription)
+                }
+            }
+            
+            logger.log(urlRequest)
+        }
+        
+        XCTAssertFalse(logged.isEmpty)
+    }
+    
+    func testLogHeaders() {
+        let headers = [
+            "X-Stream-Client": "stream-chat-swift-client-2",
+            "X-Stream-Device": "iPhone11",
+            "X-Stream-OS": "iOS13.3",
+            "Stream-Auth-Type": "anonymous"]
+        
+        logger.log(headers: headers)
+        
+        XCTAssertFalse(logged.isEmpty)
+        for (key, value) in headers {
+            XCTAssert(logged[0].contains("\(key) = \(value)"))
+        }
+    }
+    
+    func testLogQueryItems() {
+        let queryItems: [URLQueryItem] = [.init(name: "test1", value: "test1"),
+                                          .init(name: "test2", value: "test2")]
+        
+        logger.log(queryItems)
+        
+        XCTAssertFalse(logged.isEmpty)
+        XCTAssertEqual(logged[0], "URL query items:\n▫️ test1=test1\n▫️ test2=test2\n")
+    }
+    
+    func testLogURLResponse() {
+        let urlResponse = URLResponse(url: testUrl, mimeType: nil, expectedContentLength: 1, textEncodingName: nil)
+        
+        logger.log(urlResponse, data: testData)
+        
+        XCTAssertFalse(logged.isEmpty)
+        XCTAssert(logged[0].starts(with: "⬅️❔ Unknown response (6 bytes): "))
```

Comment:
@buh we should really kill all these emojis from logs; I know they are cute but anybody shipping these logs to a centralised logging facility will hate us when that breaks some Perl based data pipeline (lot of logging software / infrastructure / scripts still think there are only 255 characters)

Reply:
It's not for funny stuff. It is really easy to read needed logs and filter by emoji in Xcode.
It's configurable and emojis can be removed. For me, it's really useful and as I know @b-onc like them too.

Reply:
As long as we don't get any complaints I'm happy with emoji logging. It's a strange new thing for me but it really makes reading through logs clearer (and more fun as a side effect)

I'd prefer something more mainstream (and expected), like putting emojis per log level (instead of per log function). I've seen people using differently colored hearths (❤️, 🧡, 💛, 💚, etc...) for debug, error, info, verbose, etc...